src/tests/README.md

```diff
@@ -0,0 +1,152 @@
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+# Testing Checklist
+
+## Requirements
+- [ ] install Java 8 or higher
+- [ ] supported VSCode version
+- [ ] operating system
+  - [ ] Windows
+  - [ ] Linux
+  - [ ] MacOS
+
+## Installation
+- [ ] install extension .vsix from file
+- [ ] install extension from marketplace
+- [ ] verify that the extension is installed
+- [ ] more here...
+
+## DFDL Debugger
+- [ ] open a DFDL file
+- [ ] verify source highlighting
+- [ ] verify code completion
+- [ ] more here...
+
+## TDML
+- [ ] open a TDML file
+- [ ] Generate a TDML file from debugging a daffodil schema
+- [ ] Verify that the generated TDML file contains correct relative paths
+- [ ] Veirfy parserTestCase object has name, root, model, description
+- [ ] Verify that the infoSet and DocumentPart are correct
+- [ ] Append to a TDML file by debugging a schema and specifying append and the file to append to
+- [ ] Verify the new parserTestCase that was appended to the file
+- [ ] Execute a TDML Schema through the debugger
+- [ ] Verify that output generated matches original input
+- [ ] Make sure that these work on Windows 
+
+## Data Editor
+- [ ] open a file (use small and large files of different types)
+  - [ ] verify that the tab is the base name of the file
+  - [ ] verify that the full path name is correct (if the file name is too long, there will be a horizontal scroll bar)
+  - [ ] verify that the Disk Size and Computed Size match the actual size of the file (hover over the values to see the size in bytes)
+  - [ ] verify that the file is loaded in less than 5 seconds regardless of the size of the file
+  - [ ] verify that the detected Content Type is correct for the file
+  - [ ] verify that the header min/max button shows and hides the header
+- [ ] navigation
+  - [ ] scroll through the file, using navigation buttons
+  - [ ] scroll through the file, using the pseudo scrollbar
+  - [ ] scroll through the file, using the mouse wheel
+  - [ ] scroll through the file, using keyboard shortcuts (up, down, page up, page down, home, end)
+- [ ] display modes
+  - [ ] change the address radix (hexadecimal, decimal, octal)
+  - [ ] change the display radix (hexadecimal, decimal, octal, binary)
+- [ ] profiler
```

Comment:
## Profiler Discrepancies
### Offsets and Length Changes are `NaN`
Changed the start/end offset to `FF` and it resulted the 'Start/End Offset' and 'Length' being set to `NaN`.
<details><summary><h4>Screenshot</h4></summary>

![image](https://github.com/apache/daffodil-vscode/assets/30351915/fde9c072-3d85-4963-9a0d-94a0b1bf5154)

</details>

---

### Assert Failure on Length Change
A console error in regards to an assertion failure appears when changing the value of the 'Length' field.
<details><summary><h4>Screenshot</h4></summary>

![image](https://github.com/apache/daffodil-vscode/assets/30351915/04815ce6-16a7-4088-bfc8-87b98ae7717b)

</details>